src/main/java/bio/terra/app/model/GoogleRegion.java

```diff
@@ -82,8 +88,10 @@ public static GoogleRegion fromValueWithDefault(String text) {
     }
 
     public static boolean matchingRegionWithFallbacks(List<StorageResource> storage, GoogleRegion region) {
-        for (StorageResource resource : storage) {
-            switch (resource.getCloudResource()) {
+        List<StorageResource> gcpStorage =
+            storage.stream().filter(s -> s.getCloudPlatform() == CloudPlatform.GCP).collect(Collectors.toList());
```

Comment:
It might be clearer to put the cast here
```suggestion
            storage.stream().filter(s -> s.getCloudPlatform() == CloudPlatform.GCP).map(r -> (GoogleCloudResource) r).collect(Collectors.toList());
```

Reply:
Ah I see.  Yeah, I'd need to change the variable definition too